pkg/controller/siddhiprocess/artifact/artifacts.go

```diff
@@ -427,87 +438,87 @@ func (rsp *ReconcileSiddhiProcess) CreateOrUpdateDeployment(
 	return
 }
 
-// DeleteService delete the service specify by the user
-func (rsp *ReconcileSiddhiProcess) DeleteService(
-	serviceName string,
+// DeleteService deletes the service specified by the user
+func (k *KubeClient) DeleteService(
+	name string,
 	namespace string,
 ) (err error) {
 	service := &corev1.Service{}
-	er := rsp.client.Get(
+	er := k.Client.Get(
 		context.TODO(),
-		types.NamespacedName{Name: serviceName, Namespace: namespace},
+		types.NamespacedName{Name: name, Namespace: namespace},
 		service,
 	)
 	if er == nil {
-		err = rsp.client.Delete(context.TODO(), service)
+		err = k.Client.Delete(context.TODO(), service)
 		if err != nil {
 			return
 		}
 	}
 	return
 }
 
-// DeleteDeployment delete the deployment specify by the user
-func (rsp *ReconcileSiddhiProcess) DeleteDeployment(
-	deploymentName string,
+// DeleteDeployment deletes the deployment specify by the user
+func (k *KubeClient) DeleteDeployment(
+	name string,
 	namespace string,
 ) (err error) {
 	deployment := &appsv1.Deployment{}
-	er := rsp.client.Get(
+	er := k.Client.Get(
 		context.TODO(),
-		types.NamespacedName{Name: deploymentName, Namespace: namespace},
+		types.NamespacedName{Name: name, Namespace: namespace},
 		deployment,
 	)
 	if er == nil {
-		err = rsp.client.Delete(context.TODO(), deployment)
+		err = k.Client.Delete(context.TODO(), deployment)
 		if err != nil {
 			return
 		}
 	}
 	return
 }
 
-// DeletePVC delete the PVC specify by the user
-func (rsp *ReconcileSiddhiProcess) DeletePVC(
-	pvcName string,
+// DeletePVC deletes the PVC specify by the user
+func (k *KubeClient) DeletePVC(
```

Comment:
Do we need to delete the PVC? This does not delete the bounded PV. Normally, we keep the PVC without deleting so that the admin will delete it when required.

Reply:
Here we are the one explicitly creating the PVC. That is why here we have a method to delete in once it needed.